src/vitabel/vitals.py

```diff
@@ -924,21 +1015,67 @@ def get_channel_or_label(
             )
         return channels_or_labels[0]
 
-    def get_channel_names(self) -> list[str]:  # part of register application
-        """Return a list with the names of all channels in the recording."""
-        return self.data.channel_names
+    def get_channel_names(self, **kwargs) -> list[str]:  # part of register application
+        """Return a list with the names of all channels in the recording.
+        
+        Parameters
+        ----------
+        kwargs
+            Keyword arguments to filter the channels by.
+        
+        Returns
+        -------
+        list[str]
+            A list with the names of all channels in the recording.
+            If no channel matches the specification, an empty list is returned.
+        """
 
-    def get_label_names(self) -> list[str]:  # part of register application
-        """Returns a list with the names of all labels."""
-        return self.data.label_names
+        channels = self.data.get_channels(**kwargs)
+        
+        return [channel.name for channel in channels]
+
+    def get_label_names(self, **kwargs) -> list[str]:  # part of register application
+        """Returns a list with the names of all labels.
+
+        Parameters
+        ----------
+        kwargs
+            Keyword arguments to filter the labels by.
+            See :meth:`.get_label` for valid specifications.
 
-    def get_channel_or_label_names(self) -> list[str]:
-        """Returns a list with all names from either channels or labels."""
-        return self.get_channel_names() + self.get_label_names()
+        Returns
+        -------
+        list[str]
+            A list with the names of all labels in the recording.
+            If no label matches the specification, an empty list is returned.
+        
+        """
+
+        labels = self.data.get_labels(**kwargs)
+        
+        return [label.name for label in labels]
+
+    def get_channel_or_label_names(self, **kwargs) -> list[str]:
+        """Returns a list with all names from either channels or labels.
+        
+        Parameters
+        ----------
+        kwargs
+            Keyword arguments to filter the channels and labels by.
+            See :meth:`.get_channel` and :meth:`.get_label` for valid specifications.
+
+        Returns
+        -------
+        list[str]
+            A list with the names of all channels and labels in the recording.
+            If no channel or label matches the specification, an empty list is returned.
+        """
+        return self.get_channel_names(**kwargs) + self.get_label_names(**kwargs)
 
-    def keys(self) -> list[str]:
+    def keys(self, **kwargs) -> list[str]:
```

Comment:
Usually, method aliases can simply be implemented by straightforward assignment:

```
    keys = get_channel_or_label_names
```